File: 2nd Year/4th Semester/VIVA/AOA/EXPERIMENT 01/selectionsort.c
#include<stdio.h>

int arr[] = {65, 25, 12, 22, 11};
int n = sizeof(arr)/sizeof(arr[0]);

void sortArray() {
	for(int i = 0; i < n - 1; i++) {
		int min = i;
		for(int j = i+1; j < n; j++)
			if(arr[min] > arr[j])
				min = j;
		int temp = arr[i];
		arr[i] = arr[min];
		arr[min] = temp;
	}
}

void printArray() {
	for(int i = 0; i < n; i++)
		printf("%d ", arr[i]);
}

void main() {
	printf("Unsorted Array:");
	printArray();
	printf("\nSorted Array:");
	sortArray();
	printArray();
}

File: 2nd Year/4th Semester/VIVA/AOA/EXPERIMENT 01/SelectionSort.java
//Selection.
//swapping.
//counter shift.

class SelectionSort {
	static int[] arr = {5, 4, 3, 2, 1};
	public static void main(String...args) {
		int min = 0, n = arr.length;
		for(int i = 0; i < n -1; i++) {
			min = i;
			for(int j = i+1; j < n; j++) {
				if(arr[j] < arr[min]) {
					min = j;
				}
			}
			int temp = arr[min];
			arr[min] = arr[i];
			arr[i] = temp;
		}
		for(int i = 0; i < n; i++) {
			System.out.println(arr[i] + " ");
		}
	}
}